### PR TITLE
add possibility to add multiple session parameters at one time

### PR DESCRIPTION
**Pull request by ismail1432**
| Q             | A
| ------------- | ---
| Branch?       | master for features / 2.7 up to 4.0 for bug fixes 
| Bug fix?      | no
| New feature?  | yes 
| BC breaks?    | no     
| Deprecations? | yes/no 
| Tests pass?   | yes    
| Fixed tickets | #26616   
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

This PR should allow User to add multiple session parameters at one time

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/Session/Session.php [hunk 1]
```diff
@@ -258,4 +258,16 @@ private function getAttributeBag()
     {
         return $this->storage->getBag($this->attributeName);
     }
+
+    /**
+     * Add many attributes in session at one time
+     *
+     */
+    public function add(array $datas)
```
Comment (fancyweb) [inline src/Symfony/Component/HttpFoundation/Session/Session.php]: I think the word "datas" is an English mistake.

Comment (webignition) [reply src/Symfony/Component/HttpFoundation/Session/Session.php]: Yep, "data" is already plural.

Maybe a better choice would be a world that can take a plural form?

I don't have much context here, but maybe `$items`?

Comment (robfrawley) [reply src/Symfony/Component/HttpFoundation/Session/Session.php]: Ditto. Just like you wouldn't pluralize "information", the same applies to "data".

Comment (ismail1432) [reply src/Symfony/Component/HttpFoundation/Session/Session.php]: Hi everyone,
Thanks for feedback, maybe this feature will never exist. We are speaking about benefits or not..
I will change code only if Symfony core team need the feature.

File: src/Symfony/Component/HttpFoundation/Session/Session.php [hunk 2]
```diff
@@ -258,4 +258,16 @@ private function getAttributeBag()
     {
         return $this->storage->getBag($this->attributeName);
     }
+
+    /**
+     * Add many attributes in session at one time
```
Comment (hhamon) [inline src/Symfony/Component/HttpFoundation/Session/Session.php]: Missing dot at the end of the description I think.

File: src/Symfony/Component/HttpFoundation/Session/Session.php [hunk 3]
```diff
@@ -258,4 +258,16 @@ private function getAttributeBag()
...
+     *
+     */
+    public function add(array $datas)
+    {
+        foreach ($datas as $name => $value)
+        {
```
Comment (hhamon) [inline src/Symfony/Component/HttpFoundation/Session/Session.php]: curly brace should be on the same line as `foreach`

File: src/Symfony/Component/HttpFoundation/Tests/Session/SessionTest.php [hunk 4]
```diff
@@ -219,4 +219,23 @@ public function testGetMeta()
     {
         $this->assertInstanceOf('Symfony\Component\HttpFoundation\Session\Storage\MetadataBag', $this->session->getMetadataBag());
     }
+
+    public function testAdd()
+    {
+        $datas = [
```
Comment (hhamon) [inline src/Symfony/Component/HttpFoundation/Tests/Session/SessionTest.php]: `$data`

**Comment by fabpot** [comment]
As per the discussion in the related issue, I don't think this is worth it.

**Comment by fabpot** [comment]
Closing as I don't think this is worth it.

**Pull request closed without merging**